### PR TITLE
MM-T3080 Add Cypress test for Sign in email/pwd account

### PR DESCRIPTION
**Pull request by yulyanaR**
**Summary**

Cypress test for Sign in email/pwd account MM-T3080

**Ticket Link**

https://automation-test-cases.vercel.app/test/MM-T3080

### CODE REVIEW

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Stage: @prod @smoke
+// Group: @signin_authentication
```
Comment (Willyfrog) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:11]: I believe this has to be filled by QA, so I'd remove them

**Review by Willyfrog — APPROVED** [review]
looking good minus one thing regarding how qa usually does this process.

**Comment by yulyanaR** [comment]
Fixed as per the request, waiting for approval. Thanks!

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @signin_authentication
+
+describe('SignIn Authentication', () => {
+    let testUser;
+
+    before(() => {
+        // # Create new team and users
```
Comment (saturninoabril) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:16]:
```suggestion
        // # Create new user
```

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        cy.apiInitSetup().then(({user}) => {
+            testUser = user;
+
+            cy.apiLogout();
+            cy.visit('/login');
+        });
+    });
+
+    it('MM-T3080 Sign in email/pwd account', () => {
+        // # Enter actual users email in the email field
```
Comment (saturninoabril) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:26]:
```suggestion
        // # Enter actual user's email in the email field
```

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        cy.apiGetClientLicense().then(({isLicensed}) => {
+            const loginPlaceholder = isLicensed ? 'Email, Username or AD/LDAP Username' : 'Email or Username';
+            cy.findByPlaceholderText(loginPlaceholder).clear().type(testUser.email);
+
+            // # Enter any password in the password field
```
Comment (saturninoabril) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:31]:
```suggestion
            // # Enter user's password in the password field
```

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            // # Click hamburger main menu button
+            cy.findByLabelText('main menu').should('be.visible').click();
+
+            // # Click on the logout menu
+            cy.findByText('Log Out').scrollIntoView().should('be.visible').click();
+
+            // * Check that it logout successfully and it redirects into the login page
+            cy.url().should('include', '/login');
+
+            // # Enter actual users username in the email field
```
Comment (saturninoabril) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:49]:
```suggestion
            // # Enter actual user's username in the email field
```

File: e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            cy.findByPlaceholderText(loginPlaceholder).clear().type(testUser.username);
+
+            // # Enter any password in the password field
```
Comment (saturninoabril) [inline e2e/cypress/integration/signin_authentication/login_logout_smoke_spec.js:52]:
```suggestion
            // # Enter user's password in the password field
```

**Review by saturninoabril — APPROVED** [review]
@yulyanaR Thanks for updating! I left last comments which I missed to mention last time. Approving now given those are to be updated.

**Review by prapti — APPROVED** [review]
Hi @yulyanaR! This is great! The test flow is exactly what's needed and you did it perfectly. Thank you so much for your contribution and congratulations on your first Mattermost PR 🎉

**Pull request merged**